lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -16,12 +16,13 @@ def run(&block)
           queue = build_queue
           lock = Mutex.new
 
-          @workers = thread_count.times.map do
-            Thread.new do
+          @workers = Array.new(thread_count) do
+            t = Thread.new do
               while (item = next_item(queue, lock))
                 yield item
               end
             end
+            t.abort_on_exception = true
```

Comment:
@brynary BTW I looked at the `thread` object returned from `Thread#join` but so far didn't see a quick and useful way to interrogate whether there'd been an exception. I also think there could be some slight performance advantages to aborting early.

WDYT about the current implementation instead?